corehq/apps/integration/kyc/views.py

```diff
@@ -106,16 +104,20 @@ def _parse_row(self, row_obj):
             'country',
         )
         system_fields = (
-            'kyc_is_verified',
+            'kyc_verification_status',
             'kyc_last_verified_at',
         )
-        for field in user_fields:
-            if field not in user_data or user_data[field] in ('', None):
-                row_data['has_invalid_data'] = True
-                continue
-            row_data[field] = user_data[field]
-        for field in system_fields:
-            row_data[field] = user_data.get(field)
+        for field in (user_fields + system_fields):
+            value = None
+            try:
+                value = row_obj[field]
+            except KeyError:
+                pass
+            finally:
+                if value in ['', None] and field in user_fields:
+                    row_data['has_invalid_data'] = True
+                else:
+                    row_data[field] = value
```

Comment:
If you define a `get()` method on `KycUser` (because duck-typing dictionaries is fun) like so
```python
    def get(self, item, default=None):
        try:
            return self[item]
        except KeyError:
            return default
```
then you can drop the `try ... except` block here.
```suggestion
            value = row_obj.get(field)
            if not value and field in user_fields:
                row_data['has_invalid_data'] = True
            else:
                row_data[field] = value
```

Reply:
Interesting.

Maybe, It looks like @ajeety4 specifically wants to identify if the key is not present, since `None` is a valid value. A check for key being present would have made that intent clear.

Reply:
>  It looks like @ajeety4 specifically wants to identify if the key is not present

Not really. Here, key not being present and value being `None` both means row has invalid data for `user_fields`. 
Now I realise below would have been better

```
  try:
      value = row_obj[field]
  except KeyError:
      value = None
  finally:
     <code>
```